src/lib/config.ts

```diff
@@ -2,18 +2,49 @@ import { resolveConfig as viteResolveConfig } from "vite";
 
 import type { FrameworkConfig } from "../config.js";
 
+type DeepRequired<T> = T extends (...args: any[]) => any
+  ? T
+  : T extends object
+  ? { [P in keyof T]-?: DeepRequired<T[P]> }
+  : T;
+
+const splitHTML = (htmlStr: string): readonly [string, string] => {
+  const startStr = "<!--placeholder-->";
+  const endStr = "<!--/placeholder-->";
+  const splitted = htmlStr.split(new RegExp(startStr + "[\\s\\S]*" + endStr));
+  if (splitted.length !== 2) {
+    throw new Error("Failed to split HTML");
+  }
+  return [splitted[0] + startStr, endStr + splitted[1]];
+};
+
+const getFallback = (id: string) => {
+  if (id.endsWith("#Waku_SSR_Capable_Link")) {
+    return "waku/server#ClientOnly";
+  }
+  return "waku/server#ClientFallback";
+};
```

Comment:
I tried to create a full SSR (= also render client components into HTML on server) example, with configuring `getFallback`, but there are a few issues and I gave it up doing in this PR:
- We need a module graph in DEV, which we don't have in our current architecture.
- In production mode, it should be easier, but the configuration would be tricker.
- What's really blocking is probably as we load React in a separate worker, we get "Invalid hook call." error. I don't think we can do SSR in the same worker, so we'd need a new separate worker and duplicate the work. That doesn't feel very RSC-first framework.

We will see it again in the future, maybe after we migrate to `react-server-dom-vite`, which is not yet planned.